WikiPreview.Fluent.Plugin/WikiPreviewSearchApp.cs

```diff
@@ -0,0 +1,166 @@
+                        managerInstance.StartNewProcess(url);
+                        return new ValueTask<IHandleResult>(new HandleResult(true, false));
+                    }
+
+                case var type when type == typeof(CopyUrlSearchOperation):
+                    {
+                        string wikiUrl = wikiRootUrl + searchResult.DisplayedName;
+                        TextCopy.Clipboard.SetText(wikiUrl);
+                        return new ValueTask<IHandleResult>(new HandleResult(true, false));
+                    }
+
+                case var type when type == typeof(GoogleSearchOperation):
+                    {
+                        string url = "https://www.google.com/search?q=" + searchResult.DisplayedName;
+                        managerInstance.StartNewProcess(url);
+                        return new ValueTask<IHandleResult>(new HandleResult(true, false));
+                    }
+            }
+
+            return new(new HandleResult(true, false));
+        }
+
+        public ValueTask LoadSearchApplicationAsync()
+        {
+            return ValueTask.CompletedTask;
+        }
+
+        public async IAsyncEnumerable<ISearchResult> SearchAsync(SearchRequest searchRequest, CancellationToken cancellationToken)
+        {
+            string searchedTag = searchRequest.SearchedTag;
+            string searchedText = searchRequest.SearchedText;
+            searchedText = searchedText.Trim();
+
+            if (string.IsNullOrWhiteSpace(searchedTag) && !searchedTag.Equals(WikiSearchTagName, StringComparison.OrdinalIgnoreCase) && string.IsNullOrWhiteSpace(searchedText))
+            {
+                yield break;
+            }
+
+            QueryConfiguration queryConfiguration = new() { SearchTerm = searchedText, WikiNameSpace = 0, ImageSize = 100, ResultsCount = 8, SentenceCount = 8 };
+            string url = GetFormattedURL(queryConfiguration);
+            using var httpClient = new HttpClient();
+            Channel<WikiPreviewSearchResult> channel = Channel.CreateUnbounded<WikiPreviewSearchResult>();
+
+            JsonSerializerOptions serializerOptions = new() { PropertyNameCaseInsensitive = true };
+            _ = httpClient.GetFromJsonAsync<Wiki>(url, serializerOptions, cancellationToken).ContinueWith(task =>
+             {
+                 if (task.IsCompletedSuccessfully && task.Result != null)
+                 {
+                     _ = task.Result.Query.Pages.ParallelForEachAsync(async entry =>
+                     {
+                         string resultName = entry.Value.Extract;
+                         string displayedName = entry.Value.Title;
+                         double score = displayedName.SearchDistanceScore(searchedText);
+                         string pageID = entry.Value.PageId.ToString();
+                         string url = wikiRootUrl + displayedName;
+                         string resultType = "";
```

Comment:
Why result type is empty? I suggest you using a const value inside your search result